mozilla/foundation.mozilla.org
# make sure we only check for name duplicates in the snippet's locale

Closes https://github.com/mozilla/foundation.mozilla.org/issues/8148 by making sure to check for locale, if we're dealing with localised data.

I will also be filing a bug with wagtail over the `locale` property, because that should always resolve to something, not need two completely different code paths despite running wagtail with localization turned on.

# Testing

- checkout `main`
- run `inv copy-stage-db`
- once done, run `docker-compose up` and navigate to http://localhost:8000/cms
- go to snippets -> buyer's guide category snippets
- click "edit" for the "Valentine's day" entry, and then immediately save it without changing anything
- this should fail
- keep the browser open
- checkout this branch
- reload the browser tab
- this should now succeed
- change the "Valentine's day" category's name to "Wearables" and save
- this should still fail (as there is already a category by that name)
